From: SJ Park <sj@kernel.org>
Date: Tue, 4 Aug 2026 07:42:29 -0700
Subject: [PATCH] mm/damon/core: error damos_commit_quota_goal() for zero
 target_value

Otherwise, it could cause divide-by-zero in damos_quota_goal().

Zero target value is anyway useless, at least for now, since it cannot
be negative.  Quota goal will always be evaluated as achieved or over
achieved, make quotas stay or reduced.  Eventually it will only make
quota zero.

In future, we might introduce quota tuning to work in reverse.  Still,
even in this case, it is evaluated as achieved or under achieved.  Quota
will only get increased, make no sense.

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index c52e5bdbe797d..170e5b93c6167 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1193,15 +1193,18 @@ static void damos_commit_quota_goal_union(
 	}
 }
 
-static void damos_commit_quota_goal(
+static int damos_commit_quota_goal(
 		struct damos_quota_goal *dst, struct damos_quota_goal *src)
 {
+	if (!src->target_value)
+		return  -EINVAL;
 	dst->metric = src->metric;
 	dst->target_value = src->target_value;
 	if (dst->metric == DAMOS_QUOTA_USER_INPUT)
 		dst->current_value = src->current_value;
 	/* keep last_psi_total as is, since it will be updated in next cycle */
 	damos_commit_quota_goal_union(dst, src);
+	return 0;
 }
 
 /**
-- 
2.47.3

